### ISSUE TITLE
E1136: Unsubscriptable-object Error

### ISSUE THREAD

**Issue by kzmsdkn** [pylint-dev/pylint#9350]
### Bug description

```python
from abc import ABC, abstractmethod
from typing import TypeVar

from src.internal.core import entities, dtos


Entity = TypeVar("Entity")
CreateDto = TypeVar("CreateDto")
UpdateDto = TypeVar("UpdateDto")


class BaseDao[Entity, CreateDto, UpdateDto](ABC):
    @abstractmethod
    def create(self, *_, **kwargs: CreateDto) -> Entity | None:
        return NotImplemented

    @abstractmethod
    def find_all(self) -> list[Entity] | None:
        return NotImplemented

    @abstractmethod
    def find_by_id(self, uid: str) -> Entity | None:
        return NotImplemented

    @abstractmethod
    def update(self, uid: str, **kwargs: UpdateDto) -> Entity | None:
        return NotImplemented

    @abstractmethod
    def delete(self, uid: str) -> None:
        return NotImplemented


class UserDao(BaseDao[entities.User, dtos.CreateUserDto, dtos.UpdateUserDto]):
    pass
```

### Command used

```shell
vscode pylint extension
```

### Pylint output

```python
Value 'BaseDao' is unsubscriptablePylint[E1136:unsubscriptable-object](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/unsubscriptable-object.html)
```

### Expected behavior

Is this my code wrong, or pylint is not detecting the code correctly?
If I add more methods to UserDao, the error's gone. However, I just want to prevent redundant abstract classes.

### Pylint version

```shell
v2023.11.13481007
```

### OS / Environment

macos

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Will be closing as a duplicate of a slimmer example reported and triaged in #9406
